src/components/BookmarksPanel.tsx

```diff
@@ -0,0 +1,116 @@
+/** ****************************************************************************
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************** */
+
+import { observer } from 'mobx-react-lite';
+import moment from 'moment';
+import React from 'react';
+import { getTimestampAsNumber } from '../helpers/date';
+import { isEventMessage } from '../helpers/event';
+import { createBemElement } from '../helpers/styleCreators';
+import { useSelectedStore } from '../hooks';
+import { EventAction } from '../models/EventAction';
+import { EventMessage } from '../models/EventMessage';
+
+interface SavedItem {
+	value: EventMessage | EventAction;
+	type: 'pinned-message' | 'event';
+}
+
+const BookmarksPanel = () => {
+	const selectedStore = useSelectedStore();
+	const savedItems: SavedItem[] = React.useMemo(() => {
+		return ([...selectedStore.attachedMessages, ...selectedStore.pinnedMessages] as (
+			| EventAction
+			| EventMessage
+		)[])
+			.map(item => ({
+				value: item,
+				type: isEventMessage(item) ? ('pinned-message' as 'pinned-message') : ('event' as 'event'),
+			}))
+			.sort((firstItem, secondItem) => {
+				const getItemTimestamp = (item: SavedItem) =>
+					getTimestampAsNumber(
+						item.type === 'event'
+							? (item.value as EventAction).startTimestamp
```

Comment:
we have type guards for Events and Message, (isEventMessage, isEventAction) why don't we check value? either use it or create a new type guard based on item type